firstvoices/backend/management/commands/_helper.py

```diff
@@ -0,0 +1,56 @@
+from elasticsearch.exceptions import NotFoundError
+from elasticsearch.helpers import bulk
+from elasticsearch_dsl.connections import connections
+
+from backend.models.dictionary import DictionaryEntry
+from backend.search.indices.dictionary_entry_document import (
+    DictionaryEntryDocument,
+    dictionary_entries,
+)
+from backend.search.utils.object_utils import (
+    get_notes_text,
+    get_translation_and_part_of_speech_text,
+)
+
+
+def rebuild_index(index, index_document):
+    es = connections.get_connection()
+
+    # Delete index
+    delete_index(index)
+
+    # Initialize new index
+    index_document.init()
+
+    # Add documents
+    if index == dictionary_entries:
+        bulk(es, dictionary_entry_iterator())
+
+    # Songs and stories to be added later
+
+
+def delete_index(index):
+    try:
+        index.delete()
+    except NotFoundError:
+        print("Current index not found for deletion. Creating a new index.")
+
+
+def dictionary_entry_iterator():
+    queryset = DictionaryEntry.objects.all()
+    for entry in queryset:
+        (
+            translations_text,
+            part_of_speech_text,
+        ) = get_translation_and_part_of_speech_text(entry)
```

Comment:
It took less than a 4 seconds for <10 documents to be indexed, since the deletion, creation takes a few seconds as well. The bulk API to index should handle large queryset size, and since iterator is being used, memory should not be a problem as well.
The only place where I feel this can be optimized further is these 2 lines where we are fetching translation and notes after fetching the object. 
Leaving this note here for reference.